packages/tools/src/user-interaction.ts

```diff
@@ -0,0 +1,88 @@
+import { confirm, select, text } from '@clack/prompts'
+import { tool } from 'ai'
+import { z } from 'zod'
+
+type Spinner = {
+  start: (msg?: string) => void
+  stop: (msg?: string, code?: number) => void
+  message: (msg?: string) => void
+}
+
+export const makeInteractiveToolset = (s: Spinner) => {
+  return {
+    askQuestion: askQuestion(s),
+    getConfirmation: getConfirmation(s),
+    presentOptions: presentOptions(s),
+    concludeTask,
+  }
+}
+
+export const askQuestion = (s: Spinner) =>
+  tool({
+    description: 'Interact with user to as him a question',
+    parameters: z.object({
+      question: z.string().describe('What do you want to ask'),
+    }),
+    execute: async ({ question }) => {
+      s.stop('A question:')
+      const response = await text({
+        message: question,
+        validate: (value) => (value.length > 0 ? undefined : 'Please provide a valid answer.'),
+      })
+      s.start()
+      return response
+    },
+  })
+
+export const getConfirmation = (s: Spinner) =>
```

Comment:
nit: to keep this inline with other tools, this should be `confirmOperation`. You may need to update parameter name.

Reply:
will do